scodec/shared/src/main/scala/fs2/scodec/ByteVectorChunk.scala

```diff
@@ -0,0 +1,74 @@
+package fs2
+package scodec
+
+import scala.reflect.{ClassTag, classTag}
+import _root_.scodec.bits.ByteVector
+
+final class ByteVectorChunk private (val toByteVector: ByteVector)
+    extends MonomorphicChunk[Byte]
+{
+  def apply(i: Int): Byte =
+    toByteVector(i)
+
+  def copyToArray[B >: Byte](xs: Array[B], start: Int): Unit =
+    xs match {
+      case byteArray: Array[Byte] =>
+        toByteVector.copyToArray(byteArray, start)
+      case _ =>
+        toByteVector.toArray.iterator.copyToArray(xs, start)
```

Comment:
Not that it matters much, but why not just replace this with `iterator.copyToArray(xs, start)`?  I mean the slow-path, non-`Byte` case, obviously.